File: snippets/csharp/thread-messages_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;
// using System.Linq;

// Create conversation thread
var thread = new List<ChatMessage>
{
    new SystemMessage { Contents = new List<AIContent> { new TextContent { Text = "You are a helpful assistant." } } },
    new UserMessage { Contents = new List<AIContent> { new TextContent { Text = "Hello!" } } },
    new AgentMessage { Contents = new List<AIContent> { new TextContent { Text = "Hi! How can I help?" } } }
};

// Serialize thread
var serializer = new MessageSerializer();
var threadXml = thread.Select(msg => serializer.Serialize(msg)).ToList();

Console.WriteLine($"Thread length: {threadXml.Count} messages");

File: docs/snippets/python/thread-messages_main.py
# from microsoft.agents.xml.models import SystemMessage, ChatMessage, AgentMessage, TextContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Create conversation thread
# thread = [
#     SystemMessage(contents=[TextContent(text="You are a helpful assistant.")]),
#     ChatMessage(role="user", contents=[TextContent(text="Hello!")]),
#     AgentMessage(role="assistant", contents=[TextContent(text="Hi! How can I help?")])
# ]
#
# # Serialize thread
# serializer = MessageSerializer()
# thread_xml = [serializer.serialize(msg) for msg in thread]
#
# print(f"Thread length: {len(thread_xml)} messages")